### DISCUSSION TITLE
Why 504 occurs after using React context?

### DISCUSSION THREAD

**Discussion by MattinaYang** [vercel/community#1846] [Help] [answered]
### Summary

I used React context in my project which deployed by Vercel and then all my apis returned 504. I wonder if Vercel not support React context? Or is there something restricted about context by Vercel?

**Comment by MattinaYang** [comment]
![image](https://user-images.githubusercontent.com/78125933/226828369-e75b18cd-a791-496c-84b9-56bc935ffc44.png)

**Comment by amyegan** [comment]
You may be able to find more detailed information about the error in the [Runtime Logs](https://vercel.com/docs/concepts/observability/runtime-logs#). If you have a public repo or [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that you can share here, that will let us see the problem more clearly so we can help you find the cause

**Reply by MattinaYang** [reply]
Thx for your reply Firstly! I tried several times and found that context is not the point causing 504, but ReactNode is. Here is my code:
```
const Header = ({ style = {} }: Props) => {
  // read from context
  const { isHeaderHidden, setIsHeaderHidden } = useHeaderContext()

  const [suffix, setSuffix] = useState(true)

  const clickSuffixHandler = () => {
    if (isHeaderHidden) {
      setIsHeaderHidden(false)
      setSuffix(false)
    } else {
      setIsHeaderHidden(true)
      setSuffix(true)
    }
  }

  return (
    <div className={styles['root']} style={style}>
      <div
        className={styles['header-wrapper']}
        style={isHeaderHidden ? { height: 0, border: 'none' } : {}}
      >
        <div
          className={styles['header']}
          style={{ visibility: isHeaderHidden ? 'hidden' : 'visible' }}
        >
          // ...
        </div>
      </div>

      <div
        className={styles['suffix']}
        onClick={clickSuffixHandler}
      >
        // below is the key code to cause problem.
        // if use this ⬇️ , apis will return 200, everything is ok
        {suffix}
        // but, if use this ⬇️ , apis will return 504
        {/* {
          suffix 
          ? <KeyboardArrowDown /> 
          : <KeyboardArrowUp />
        } */}
      </div>
    </div>
  )
}
```

**Reply by amyegan** [reply]
It looks like the Gateway Timeout error is coming from `KeyboardArrowDown` or `KeyboardArrowUp`. Are either of them calling an API or altering data that could affect API calls at other points in your code?

**Reply by MattinaYang** [reply]
Yep KeyboardArrowDown or KeyboardArrowUp caused 504 error. But they're just svg icons provided by Mui , and there won't be any 504 with the two icons when in development env. I have no idea why 504 occurs in production env :(

**Reply by amyegan** [reply]
I'm sorry you're still facing this problem, @Sheepeer. If there is not much code being run in the icon components, then that wouldn't make sense as the reason for the timeout error. I searched for some other possibilities:

- [bad import in _app.js](https://www.reddit.com/r/nextjs/comments/112465r/comment/j8ju0z7/?utm_source=share&utm_medium=web2x&context=3)
- unanticipated logic or API calls from [Material Icons](https://mui.com/material-ui/material-icons/) and [SVGIcon](https://mui.com/material-ui/api/svg-icon/)
- imported causing function to take longer than 10 seconds leading to [execution timeout](https://vercel.com/docs/concepts/limits/overview#serverless-function-execution-timeout) error

If you can share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), I could take a closer look to help you find the cause

**Reply by MattinaYang** [reply, marked as answer]
@redblood639 @amyegan  This issue fixed!! There're two ways to import mui icons as below, the first way will cause 504 while the second way works well. I just changed to the second way and the issue gone 😃 But I still don't know why 504 occurs with the first way 
```
// the first way
import { KeyboardArrowDown, KeyboardArrowUp } from "@mui/icons-material";
// the second way
import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
import KeyboardArrowUpIcon from "@mui/icons-material/KeyboardArrowUp";
```
Thank U all !!!!

**Reply by MattinaYang** [reply]
Oh I just found that only import icons by the first way in my Header component will cause 504, if import in other component, there will be nothing wrong. Anyway, I fill so confused because I don't think there is any relationship between importing icons and APIs return 504.  Looking forward to your help, many thanks!!
